src/app/controllers/quotes-controller.ts

```diff
@@ -0,0 +1,93 @@
+import _quotes from '../../data/quotes.json';
+import { IQuote } from '../models/quotes.model';
+import { v4 as uuidv4 } from 'uuid';
+
+const generateQuotes = () => {
+    let num = 0;
+    return _quotes.map(quote => {
+        num++;
+        return {
+            ...quote,
+            id: num.toString(),
+            isDeleted: false,
+            tags: quote.tags?.split(', '),
+        };
+    });
+}
+const QUOTES: IQuote[] = generateQuotes();
+
+
+const getQuotesList = (req, res) => {
+    res.json(QUOTES);
+}
+
+const getRandomQuote = (req, res) => {
+    const randomNumber = Math.floor(Math.random() * QUOTES.length);
+    res.json(QUOTES[randomNumber]);
+}
+
+const getRandomFilteredQuote = (req, res) => {
+    const { tag, text } = req.query;
+    const quotes = QUOTES.filter(quote => {
+        if (tag && text) {
+            return quote.tags?.includes(tag) && quote.text.includes(text);
+        } else if (tag) {
+            return quote.tags?.includes(tag);
+        } else if (text) {
+            return quote.text.includes(text);
+        }
+    });
+    const randomNumber = Math.floor(Math.random() * quotes.length);
+    return quotes[randomNumber];
+}
+
+const addQuote = (req, res) => {
+    const quote = req.body;
+    const newQuote = {
+        ...quote,
+        id: uuidv4(),
+        createdAt: new Date().toISOString(),
+        isDeleted: false,
+    }
+    QUOTES.push(newQuote);
+    res.status(201).json(newQuote);
```

Comment:
Use constants for status codes, e.g. https://www.npmjs.com/package/http-status-codes